timelock/program/src/processor/process_init_timelock_set.rs

```diff
@@ -94,27 +97,83 @@ pub fn process_init_timelock_set(
     assert_rent_exempt(rent, signatory_validation_account_info)?;
     assert_rent_exempt(rent, voting_validation_account_info)?;
 
-    // Cheap computational and stack-wise calls for initialization checks, no deserialization req'd
-    assert_cheap_mint_initialized(admin_mint_account_info)?;
-    assert_cheap_mint_initialized(voting_mint_account_info)?;
-    assert_cheap_mint_initialized(yes_voting_mint_account_info)?;
-    assert_cheap_mint_initialized(no_voting_mint_account_info)?;
-    assert_cheap_mint_initialized(signatory_mint_account_info)?;
-    assert_cheap_mint_initialized(source_mint_account_info)?;
+    // Cheap computational and stack-wise calls for initialization checks, no deserialization required
+    assert_mint_initialized(signatory_mint_account_info)?;
+    assert_mint_initialized(admin_mint_account_info)?;
```

Comment:
With so many reads off the data in the mint at this point (initialization, authority, AND decimals) it makes more sense I think now to just do let mint: Mint = assert_initialized(...)?; and put the object in memory and do checks with that. Certainly is cleaner code! That being said, if you feel strongly about keeping it like this I'm not gonna hold you to it. I don't mind it.